e2e/utils.go

```diff
@@ -82,6 +90,150 @@ var (
+	res, err := strconv.Atoi(stdOut)
+	if err != nil {
+		return fmt.Errorf("failed to convert string to int: %v", stdOut)
+	}
+	if res != count {
+		return fmt.Errorf("expected omap object count %d, got %d", count, res)
+	}
+
+	return nil
+}
+
+// validateOmapCount validates no of OMAP entries on the given pool.
+// Works with Cephfs and RBD drivers and mode can be snapsType or volumesType.
+// nolint:cyclop // TODO: reduce complexity
+func validateOmapCount(f *framework.Framework, count int, driver, pool, mode string) {
+	var (
+		radosLsCmd           string
+		radosLsCmdFilter     string
+		radosLsKeysCmd       string
+		radosLsKeysCmdFilter string
+	)
+	switch strings.ToLower(mode) {
+	case volumesType:
+		switch strings.ToLower(driver) {
+		case cephfsType:
+			radosLsCmd = fmt.Sprintf("rados ls --pool=%s --namespace csi", pool)
+			radosLsCmdFilter = fmt.Sprintf("rados ls --pool=%s --namespace csi | grep -v default | grep -c ^csi.volume.", pool)
+			radosLsKeysCmd = fmt.Sprintf("rados listomapkeys csi.volumes.default --pool=%s --namespace csi", pool)
+			radosLsKeysCmdFilter = fmt.Sprintf("rados listomapkeys csi.volumes.default --pool=%s --namespace csi | wc -l", pool)
+		case rbdType:
+			radosLsCmd = fmt.Sprintf("rados ls %s", rbdOptions(pool))
+			radosLsCmdFilter = fmt.Sprintf("rados ls %s | grep -v default | grep -c ^csi.volume.", rbdOptions(pool))
+			radosLsKeysCmd = fmt.Sprintf("rados listomapkeys csi.volumes.default %s", rbdOptions(pool))
+			radosLsKeysCmdFilter = fmt.Sprintf("rados listomapkeys csi.volumes.default %s | wc -l", rbdOptions(pool))
+		default:
+			e2elog.Failf("unknown driver type: %s", driver)
+		}
+	case snapsType:
+		switch strings.ToLower(driver) {
+		case cephfsType:
+			radosLsCmd = fmt.Sprintf("rados ls --pool=%s --namespace csi", pool)
+			radosLsCmdFilter = fmt.Sprintf("rados ls --pool=%s --namespace csi | grep -v default | grep -c ^csi.snap.",
+				pool)
+			radosLsKeysCmd = fmt.Sprintf("rados listomapkeys csi.snaps.default --pool=%s --namespace csi", pool)
+			radosLsKeysCmdFilter = fmt.Sprintf("rados listomapkeys csi.snaps.default --pool=%s --namespace csi | wc -l",
+				pool)
+		case rbdType:
+			radosLsCmd = fmt.Sprintf("rados ls %s", rbdOptions(pool))
+			radosLsCmdFilter = fmt.Sprintf("rados ls %s | grep -v default | grep -c ^csi.snap.", rbdOptions(pool))
+			radosLsKeysCmd = fmt.Sprintf("rados listomapkeys csi.snaps.default %s", rbdOptions(pool))
+			radosLsKeysCmdFilter = fmt.Sprintf("rados listomapkeys csi.snaps.default %s | wc -l", rbdOptions(pool))
+		default:
+			e2elog.Failf("unknown driver type: %s", driver)
+		}
+	default:
+		e2elog.Failf("unknown mode type: %s", mode)
+	}
+	stdOut, stdErr, err := execCommandInToolBoxPod(f, radosLsCmdFilter, rookNamespace)
+	if err != nil || stdErr != "" {
+		if !strings.Contains(err.Error(), exitOneErr) {
+			e2elog.Failf("failed to execute rados command '%s' : err=%v stdErr=%s", radosLsCmdFilter, err, stdErr)
+		}
+	}
+	var err0 error
```

Comment:
we dont commonly use `err0` variable. this can be renamed to something else like `cmdErr` etc.

Reply:
Sure

Reply:
@Madhu-1 Done now